File: spring-event/src/main/java/com/study/springevent/domain/order/controller/OrderController.java
package com.study.springevent.domain.order.controller;

import com.study.springevent.domain.cart.domain.Cart;
import com.study.springevent.domain.order.service.OrderService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RequestMapping("order")
@RestController
public class OrderController {

    private final OrderService orderService;

    @PostMapping
    public ResponseEntity<?> registerOrder() {
        return ResponseEntity.ok(orderService.registerOrder(new Cart("이지은")).getOrderId());
    }
}

File: spring-amqp/order/api/src/main/kotlin/study/rabbitmq/order/api/order/api/OrderController.kt
package study.amqp.producer.order.api

import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import study.amqp.producer.order.api.request.OrderPaidRequest
import study.amqp.producer.order.service.OrderService

@RestController
@RequestMapping("/api/order")
class OrderController(
    private val orderService: OrderService,
) {

    @PostMapping("/paid")
    fun paid(@RequestBody orderDto: OrderPaidRequest) {
        orderService.paidOrder()
    }
}